### PR TITLE
python3Packages.flask-restful: drop nose dependency

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by MikaelFangel** [comment]
I unfortunately don't know what the plans for changedetection-io is.

File: pkgs/development/python-modules/flask-restful/default.nix [hunk 1]
```diff
@@ -30,7 +31,15 @@ buildPythonPackage rec {
   # conditional so that overrides are easier for web applications
   patches =
     lib.optionals (lib.versionAtLeast werkzeug.version "2.1.0") [ ./werkzeug-2.1.0-compat.patch ]
-    ++ lib.optionals (lib.versionAtLeast flask.version "3.0.0") [ ./flask-3.0-compat.patch ];
+    ++ lib.optionals (lib.versionAtLeast flask.version "3.0.0") [ ./flask-3.0-compat.patch ]
+    ++ [
+      # replace use nose by pytest: https://github.com/flask-restful/flask-restful/pull/970
+      (fetchpatch2 {
+        url = "https://github.com/flask-restful/flask-restful/commit/6cc4b057e5450e0c84b3ee5f6f7a97e648a816d6.patch?full_index=1";
```
Comment (gador) [inline pkgs/development/python-modules/flask-restful/default.nix:38]:
```suggestion
        url = "https://github.com/flask-restful/flask-restful/commit/6cc4b057e5450e0c84b3ee5f6f7a97e648a816d6.patch";
```
or is there a reason to use the full index? (If you decide to change this, please remember to change the `hash` as well)

Comment (Sigmanificient) [reply pkgs/development/python-modules/flask-restful/default.nix:38]: @emilazy wanted me to use them

Comment (emilazy) [reply pkgs/development/python-modules/flask-restful/default.nix:38]: See https://github.com/NixOS/nixpkgs/issues/266556; GitHub patches are not reproducible once repos grow big enough for the hash lengths to change, and this has actually bit us. This isn’t a problem on most repositories but I’d still consider it best practice to use `?full_index=1` until we have a `fetchpatch3` that handles this correctly. (It’s on my infinite to‐do list…)

Sorry for not reviewing these nose PRs for a few days. I will get back to it soon.

**Comment by Pizmovc** [comment]
I can confirm that  `changedetection-io` (which depends on this package) can now be built, while without this PR it can not be :+1:

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 330876` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages built:
</summary>
<ul>
<li>changedetection-io</li>
<li>changedetection-io.dist</li>
<li>nixpkgs-manual</li>
<li>python311Packages.flask-restful</li>
<li>python311Packages.flask-restful.dist</li>
<li>python312Packages.flask-restful</li>
<li>python312Packages.flask-restful.dist</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
nixpkgs-manual:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

</details>
python311Packages.flask-restful.dist:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/adzzwzaqysbhwxjmc7k2dgka2l3y2464-python3.11-flask-restful-0.3.10-dist' from 'daemon'
When evaluating attribute ‘python311Packages.flask-restful.dist’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/flask-restful/default.nix:76:5:

```
   |
76 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python311Packages.flask-restful:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/adlp6hq0ff9h6ki67gqpl82gsg1i35hv-python3.11-flask-restful-0.3.10' from 'daemon'
When evaluating attribute ‘python311Packages.flask-restful’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/flask-restful/default.nix:76:5:

```
   |
76 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python312Packages.flask-restful.dist:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/j9254kby615b2mj6qdj7m2kn112sfmwj-python3.12-flask-restful-0.3.10-dist' from 'daemon'
When evaluating attribute ‘python312Packages.flask-restful.dist’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/flask-restful/default.nix:76:5:

```
   |
76 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python312Packages.flask-restful:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/c0czn83fqkm45awxyjiyldx8wxkwyai4-python3.12-flask-restful-0.3.10' from 'daemon'
When evaluating attribute ‘python312Packages.flask-restful’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/flask-restful/default.nix:76:5:

```
   |
76 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
</details>

**Review by gador — APPROVED** [review]

**Comment by emilazy** [comment]
This didn’t actually remove the nose dependency.

**Comment by Sigmanificient** [comment]
I forget to remove it in the arguments! :disappointed_relieved:

**Comment by emilazy** [comment]
No worries; feel free to open another PR for it or I might get around to it if not.

**Pull request merged**